docs/generated/sql/aggregates.md

```diff
@@ -85,93 +85,93 @@
+<tr><td><a name="max"></a><code>max(arg1: <a href="decimal.html">decimal</a>) &rarr; anyelement</code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
 </span></td></tr>
-<tr><td><a name="max"></a><code>max(arg1: <a href="float.html">float</a>) &rarr; <a href="float.html">float</a></code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
+<tr><td><a name="max"></a><code>max(arg1: <a href="float.html">float</a>) &rarr; anyelement</code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
 </span></td></tr>
-<tr><td><a name="max"></a><code>max(arg1: <a href="inet.html">inet</a>) &rarr; <a href="inet.html">inet</a></code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
+<tr><td><a name="max"></a><code>max(arg1: <a href="inet.html">inet</a>) &rarr; anyelement</code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
 </span></td></tr>
-<tr><td><a name="max"></a><code>max(arg1: <a href="int.html">int</a>) &rarr; <a href="int.html">int</a></code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
+<tr><td><a name="max"></a><code>max(arg1: <a href="int.html">int</a>) &rarr; anyelement</code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
 </span></td></tr>
-<tr><td><a name="max"></a><code>max(arg1: <a href="interval.html">interval</a>) &rarr; <a href="interval.html">interval</a></code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
+<tr><td><a name="max"></a><code>max(arg1: <a href="interval.html">interval</a>) &rarr; anyelement</code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
 </span></td></tr>
-<tr><td><a name="max"></a><code>max(arg1: <a href="string.html">string</a>) &rarr; <a href="string.html">string</a></code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
+<tr><td><a name="max"></a><code>max(arg1: <a href="string.html">string</a>) &rarr; anyelement</code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
 </span></td></tr>
-<tr><td><a name="max"></a><code>max(arg1: <a href="time.html">time</a>) &rarr; <a href="time.html">time</a></code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
+<tr><td><a name="max"></a><code>max(arg1: <a href="time.html">time</a>) &rarr; anyelement</code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
 </span></td></tr>
-<tr><td><a name="max"></a><code>max(arg1: <a href="timestamp.html">timestamp</a>) &rarr; <a href="timestamp.html">timestamp</a></code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
+<tr><td><a name="max"></a><code>max(arg1: <a href="timestamp.html">timestamp</a>) &rarr; anyelement</code></td><td><span class="funcdesc"><p>Identifies the maximum selected value.</p>
```

Comment:
this looks unintended.

Reply:
well _technically_ it's true, since if you do max over a timestamp column but the table is empty, it returns null.

but yeah it's kinda annoying. i looked earlier into how to make the docgen stuff not do this and didn't have luck. but i think i have gained some XP so i'll try figuring it out again

Reply:
NULL is still typed (in Postgres!)

```

otan=# create table agg_a(a int);
CREATE TABLE
otan=# select max(a) from agg_a;
 max
-----

(1 row)

otan=# select pg_typeof(max(a)) from agg_a;
 pg_typeof
-----------
 integer
(1 row)

```

Reply:
yeah but CRDB doesn't have typed nulls! i mean i guess these docs were pretending we did before, so i do want to try to go back to how it was

Reply:
![image](https://user-images.githubusercontent.com/3646147/96050646-d59c3200-0e2e-11eb-86e3-ee7815a6f1f2.png)

i think the optimizer has typed nulls tho :D 
one day our type system will match...

Reply:
Actually, CRDB does have typed NULL support. It's just that the `pg_typeof` is not built to take advantage of it today. But it could be. The Optimizer keeps track of all static types in the expression, for example:
```
root@:26257/management_console_test> explain (opt,types) SELECT pg_typeof(NULL::int);
                     text
-----------------------------------------------
  values
   └── tuple [type=tuple{string}]
        └── function: pg_typeof [type=string]
             └── null [type=int]
```
The optimizer knows that the NULL has type `int`. But the `pg_typeof` function has no way of passing that information in. Instead of using a runtime function (which don't preserve types), we probably should just constant fold `pg_typeof` in the optimizer, since we always statically know the right type. This would break backwards-compatibility, but could allow us to more closely match PG semantics.